### ISSUE TITLE
windows 11 pro build error "fields_updates_ext.c" : no such file or directory

### ISSUE THREAD

**Issue by kyle-lunasonde** [gprMax/gprMax#316]
I'm using windows 11 pro build 22000 cleanly reinstalled 
I followed build directions for windows and watched videos for windows installation
(https://docs.gprmax.com/en/latest/include_readme.html)

Installed miniconda under admin privileges 
Using build tools for windows 2022 installed admin privileges
Set the Path and Environment Variables

In a new admin privilege window
Navigated to and activated gprMax
Verified that all files and folders had read and write access

Ran build step:  python setup.py build

Various compilations
Various  Cythonizing

Fatal compilation error:
c1: fatal error C1083: Cannot open source file: 'gprMax\fields_updates_ext.c': No such file or directory

But I can find the file manually and there does not seem to be any hidden characters in the path

As I'm both new to python (less than one month), is there an alternative method to install gprMax?

thanks

The entire sequence was as follows (long)
-----------------------------------------------------------------
(base) PS C:\WINDOWS\system32\gprMax> conda activate gprMax
(gprMax) PS C:\WINDOWS\system32\gprMax> python setup.py build
Running 'build_ext --inplace'
Compiling gprMax\fields_updates_ext.pyx because it changed.
Compiling gprMax\fractals_generate_ext.pyx because it changed.
Compiling gprMax\geometry_outputs_ext.pyx because it changed.
Compiling gprMax\geometry_primitives_ext.pyx because it changed.
Compiling gprMax\snapshots_ext.pyx because it changed.
Compiling gprMax\yee_cell_build_ext.pyx because it changed.
Compiling gprMax\yee_cell_setget_rigid_ext.pyx because it changed.
Compiling gprMax\pml_updates\pml_updates_electric_HORIPML_ext.pyx because it changed.
Compiling gprMax\pml_updates\pml_updates_electric_MRIPML_ext.pyx because it changed.
Compiling gprMax\pml_updates\pml_updates_magnetic_HORIPML_ext.pyx because it changed.
Compiling gprMax\pml_updates\pml_updates_magnetic_MRIPML_ext.pyx because it changed.
[ 1/11] Cythonizing gprMax\fields_updates_ext.pyx
[ 2/11] Cythonizing gprMax\fractals_generate_ext.pyx
[ 3/11] Cythonizing gprMax\geometry_outputs_ext.pyx
[ 4/11] Cythonizing gprMax\geometry_primitives_ext.pyx
[ 5/11] Cythonizing gprMax\pml_updates\pml_updates_electric_HORIPML_ext.pyx
[ 6/11] Cythonizing gprMax\pml_updates\pml_updates_electric_MRIPML_ext.pyx
[ 7/11] Cythonizing gprMax\pml_updates\pml_updates_magnetic_HORIPML_ext.pyx
[ 8/11] Cythonizing gprMax\pml_updates\pml_updates_magnetic_MRIPML_ext.pyx
[ 9/11] Cythonizing gprMax\snapshots_ext.pyx
[10/11] Cythonizing gprMax\yee_cell_build_ext.pyx
[11/11] Cythonizing gprMax\yee_cell_setget_rigid_ext.pyx
running build_ext
building 'gprMax.fields_updates_ext' extension
creating build
creating build\temp.win-amd64-3.10
creating build\temp.win-amd64-3.10\Release
creating build\temp.win-amd64-3.10\Release\gprMax
"C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.33.31629\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\kyle\miniconda3\envs\gprMax\lib\site-packages\numpy\core\include -IC:\Users\kyle\miniconda3\envs\gprMax\lib\site-packages\numpy\core\include -IC:\Users\kyle\miniconda3\envs\gprMax\include -IC:\Users\kyle\miniconda3\envs\gprMax\Include "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.33.31629\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt" /TcgprMax\fields_updates_ext.c /Fobuild\temp.win-amd64-3.10\Release\gprMax\fields_updates_ext.obj /O2 /openmp /w
cl : Command line warning D9025 : overriding '/W3' with '/w'
fields_updates_ext.c
c1: fatal error C1083: Cannot open source file: 'gprMax\fields_updates_ext.c': No such file or directory
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.33.31629\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2

**Comment by craig-warren** [comment]
@kyle-lunasonde I don't have a Windows 11 install at the moment, but I'll try and get one setup in the next few days and take a look at this.

**Comment by craig-warren** [comment]
@kyle-lunasonde I've just tried this with a clean install of 'Windows 11 Education N' and everything went fine. The only differences I can think (which shouldn't cause issues) are that I usually install Miniconda to my user account only, i.e. not for all users, and I always ensure my user account name does not contain spaces, as this sometimes causes issues with conda.

**Comment by kyle-lunasonde** [comment]
Everyone,
thanks for your help and response

I completely removed everything (gprMax, python, miniconda -- everything) via windows add/remove programs with a couple of hard reboots to my windows machine.

I then tried to build gprMax in another directory as I originally ended up in one of those windows-specific places.

The entire series of steps (conda-build-install-test) culminated with a working gprMax in a new directory far from the original folder. I also used admin privileges for everything and only installed miniconda for my account (as mentioned below).

The only issue I encountered was a warning that scipy did not like my numpy version and so to remove the warning I downgraded numpy to to 1.22.4 (was 1.23.1)

As of about 20 minutes ago, every thing seems to be in working order and the test

python -m gprMax user_models/cylinder_Ascan_2D.in

worked as documented.

I can only guess that there were windows-specific path permission issues as the file mentioned did exist but the installation steps could not find it or could find it but not access it.  Blame my bad luck or inexperience with python for this.

Or there was an error on my part setting up the window's compiler. In particular the steps:

Set the Path and Environment Variables - this can be done by following the instructions from Microsoft<https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=msvc-160#developer_command_file_locations>, or manually by adding a form of C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.23.28105\bin\Hostx64\x64 (this may vary according to your exact machine and installation) to your system Path environment variable.

was vague (actually it was the Microsoft instructions which were really vague) so I went with the manual modification of my system PATH to include the above path (suitably modified for the actual location of the MSVC compiler).



Lets call this issue (#316) closed.  Thanks again for your quick responses.

kyle

________________________________
